whois-rpsl/src/main/java/net/ripe/db/whois/common/ip/Ipv6Resource.java

```diff
@@ -33,6 +39,18 @@ public class Ipv6Resource extends IpInterval<Ipv6Resource> implements Comparable
     private final long endMsb;
     private final long endLsb;
 
+    private static final Pattern IPV6_SPLIT_PATTERN = Pattern.compile("(" +
+            "[0-9A-Fa-f]{1,4}(?::[0-9A-Fa-f]{1,4}){7}" +
+            "|::" +
+            "|:(?::[0-9A-Fa-f]{1,4}){1,6}" +
+            "|[0-9A-Fa-f]{1,4}:(?::[0-9A-Fa-f]{1,4}){1,5}" +
+            "|(?:[0-9A-Fa-f]{1,4}:){2}(?::[0-9A-Fa-f]{1,4}){1,4}" +
+            "|(?:[0-9A-Fa-f]{1,4}:){3}(?::[0-9A-Fa-f]{1,4}){1,3}" +
+            "|(?:[0-9A-Fa-f]{1,4}:){4}(?::[0-9A-Fa-f]{1,4}){1,2}" +
+            "|(?:[0-9A-Fa-f]{1,4}:){5}:[0-9A-Fa-f]{1,4}" +
+            "|(?:[0-9A-Fa-f]{1,4}:){1,6}:" +
```

Comment:
This is really complicated to understand. Is this the only way?

Reply:
I know: not I think that this is trying to check the correct syntax of the IPv6 as well, we don't need that. I am trying to create a new regex, but it is a bit complicated

Reply:
Please let me know if we can make this regex easier to understand